src/Components/CreateImageWizard/formComponents/TargetEnvironment.js

```diff
@@ -66,7 +66,7 @@ const TargetEnvironment = ({ label, isRequired, ...props }) => {
     if (getState()?.values?.[input.name]) {
       setEnvironment(getState().values[input.name]);
     }
-  }, []);
+  }, [getState, input.name]);
```

Comment:
I think you can remove the useEffect here and just keep the conditional.

Reply:
without this useEffect,  a lot of tests fail